jujugui/static/gui/src/test/test-init.js

```diff
@@ -0,0 +1,123 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const Keysim = require('keysim');
+const utils = require('./utils');
+
+fdescribe('init', () => {
+  let app, container, env, juju, JujuGUI;
+
+  beforeAll(done => {
+    // The module list is copied from index.html.mako and is required by
+    // init.js.
+    YUI(GlobalConfig).use([
+      'acl',
+      'analytics',
+      'changes-utils',
+      'juju-charm-models',
+      'juju-bundle-models',
+      'juju-controller-api',
+      'juju-endpoints-controller',
+      'juju-env-base',
+      'juju-env-api',
+      'juju-env-web-handler',
+      'juju-models',
+      'jujulib-utils',
+      'bakery-utils',
+      'net-utils',
+      // juju-views group
+      'd3-components',
+      'juju-view-utils',
+      'juju-topology',
+      'juju-view-environment',
+      'juju-landscape',
+      // end juju-views group
+      'io',
+      'json-parse',
+      'app-base',
+      'app-transitions',
+      'base',
+      'bundle-importer',
+      'bundle-import-notifications',
+      'node',
+      'model',
+      'app-cookies-extension',
+      'app-renderer-extension',
+      'cookie',
+      'querystring',
+      'event-key',
+      'event-touch',
+      'model-controller',
+      'FileSaver',
+      'ghost-deployer-extension',
+      'environment-change-set',
+      'relation-utils',
+      'yui-patches'], function(Y) {
+      // init.js requires the window to contain the YUI object.
+      window.yui = Y;
+      window.yui.juju.views = {utils: {}};
+      window.yui.juju.components = {};
+      juju = Y.namespace('juju');
+      Y.namespace('juju.environments');
+      // The require needs to be after the yui modules have been loaded.
+      JujuGUI = require('../app/init');
+      done();
+    });
+  });
+
+  beforeEach(() => {
+    container = utils.makeAppContainer();
+    const userClass = new window.jujugui.User({storage: getMockStorage()});
+    userClass.controller = {user: 'user', password: 'password'};
+    env = new juju.environments.GoEnvironment({
+      conn: new utils.SocketStub(),
+      ecs: new juju.EnvironmentChangeSet(),
```

Comment:
`juju.EnvironmentChangeSet` doesn't exist despite loading the `juju` namespace above.